src/components/10-atoms/input-text/index.js

```diff
@@ -0,0 +1,235 @@
+    this.onChange = () => {};
+
+    // internal properties
+    this.inputFocus = false;
+    this.wasFocused = false;
+    this.wasBlured = false;
+    this.nativeInput = { value: '' };
+    this.modelValue = '';
+    this.isControlled = false;
+    this.refId = this.id || `input-text-${createRefId()}`;
+  }
+
+  set value(val) {
+    const { isControlled } = this;
+    if (!isControlled && val !== undefined) {
+      this.isControlled = true;
+    }
+    const oldVal = this.modelValue;
+    this.modelValue = val;
+    this.requestUpdate('value', oldVal);
+  }
+
+  get value() {
+    const {
+      isControlled,
+      modelValue,
+      nativeInput: { value: nativeValue },
+    } = this;
+
+    return isControlled ? modelValue : nativeValue;
+  }
+
+  get isRequiredError() {
+    return this.required && !this.value;
+  }
+
+  get isInvalid() {
+    return !this.valid || this.isRequiredError;
+  }
+
+  get showInputError() {
+    return this.isInvalid && this.wasBlured && this.wasFocused;
+  }
+
+  get hideCheckIcon() {
+    if (!this.wasBlured && !this.wasFocused) {
+      return true;
+    }
+
+    if (this.inputFocus) {
+      return true;
+    }
+
+    return this.isInvalid;
+  }
+
+  get hideErrorMessage() {
+    return !this.error || !this.showInputError;
+  }
+
+  get showValidation() {
+    return this.validation || this.required;
+  }
+
+  handleFocus = ev => {
+    this.onFocus(ev);
+    this.inputFocus = true;
+
+    if (!this.wasFocused) {
+      this.wasFocused = true;
+    }
+  };
+
+  handleBlur = ev => {
+    // Validation should be on blur or submit
+    this.onBlur(ev);
+    this.inputFocus = false;
+
+    if (!this.wasBlured) {
+      this.wasBlured = true;
+    }
+  };
+
+  handleInput = ev => {
+    this.onChange(ev);
+
+    // are we a 'controlled' input in the React sense?
+    if (this.isControlled) {
+      // yes, set UI from model state
+      this.nativeInput.value = this.modelValue;
+    }
+  };
+
+  firstUpdated() {
+    this.nativeInput = this.querySelector('input');
+  }
+
+  render() {
+    const {
+      name,
+      required,
+      value,
+      label = '',
+      error = '',
+      type = '',
+      placeholder,
+      disabled,
+      isReact,
+
+      isControlled,
+      refId,
+    } = this;
+
+    this.isControlled = isControlled && isReact;
+
+    return html`
+      <div class="a-input-text__wrapper">
+        ${label &&
+          html`
+            <label for="${refId}" class="a-input-text__label"
+              >${label}
+              ${required
+                ? html`
+                    *
+                  `
+                : html``}</label
+            >
+          `}
+        <div class="a-input-text__input-wrapper">
+          <input
+            @input="${this.handleInput}"
+            @focus="${this.handleFocus}"
+            @blur="${this.handleBlur}"
+            id="${refId}"
+            type="${type}"
+            class="a-input-text__input ${this.showInputError
```

Comment:
this is ugly and error-prone and *exactly* what classMap was invented for... please rewrite